source/internal/_dissoc.js

```diff
@@ -0,0 +1,28 @@
+import _isInteger from './_isInteger';
+import _isArray from './_isArray';
+import remove from '../remove';
+
+/**
+ * Returns a new object that does not contain a `prop` property.
+ *
+ * @private
+ * @param {String|Number} prop The name of the property to dissociate
+ * @param {Object|Array} obj The object to clone
+ * @return {Object} A new object equivalent to the original but without the specified property
+ */
+export default function _dissoc(prop, obj) {
+  if (obj[prop] == null) {
+    return obj;
+  }
```

Comment:
That's going to be a problem:

```js
_dissoc('a', {a: null, b: 2}) //=> {a: null, b: 2}
```

I think you would need to switch to 

```js
  if (!(prop in obj)) {
```

I'm not sure I see much of a good reason for a special case, though, for when the property doesn't exist.  Although we stress that value equality is all we care about, we get complaints any time we sometimes return an object unchanged and sometimes return a new object.  Users who expect the latter behavior get frustrated by the former, thinking that Ramda handles it for them.  That consistency is perhaps a good thing.

Reply:
I changed the code in

* `dissocPath.js` https://github.com/ramda/ramda/pull/2787/files#diff-229ca62287d81f3b117695acff8e6d5fR33
* `_dissoc.js` https://github.com/ramda/ramda/pull/2787/files#diff-2ffae1703b1b496b4bef5eb8428bed6bR14

Reply:
> I'm not sure I see much of a good reason for a special case, though, for when the property doesn't exist. Although we stress that value equality is all we care about, we get complaints any time we sometimes return an object unchanged and sometimes return a new object. Users who expect the latter behavior get frustrated by the former, thinking that Ramda handles it for them. That consistency is perhaps a good thing.

In fact, I am not sure which way is better. At present, my understanding is that if the input will not be modified, then the original object should be returned. If it is modified, then the new object should be returned, just like immutable.js or ClojureScript.

Reply:
The difference with Immutable/ClojureScript is that the Ramda results are not actually immutable.  The problems people have had is that they work for a while with

```js
const newVal = R.foo(bar, oldVal);
```

and things are fine, as `foo` is returning something new.  Then something happens and there is no change, but

```js
alter(newVal)
```

then also alters `oldVal`.

While many Ramda users will never have issues with this, especially if most modifications use Ramda functions, people have had this problem.  I simply don't see any good reason to continue with it while we're changing this function.

Reply:
Yes, that's why I say I'm not sure which way is better. Because my point of view is based on my understanding and habits, may not suitable for other Ramda users. So I listen to you, I believe that at least this will ensure that these two functions are designed to be consistent with Ramda's overall design philosophy.

So you hope assoc/dissoc can always return a new object, no matter it will or will not be modified right?

Reply:
Yes, exactly.

Reply:
Hi @CrossEye, how about now 😆